src/serverdlg.cpp

```diff
@@ -539,6 +571,30 @@ void CServerDlg::OnStopRecorder()
     UpdateRecorderStatus ( QString::null );
 }
 
+void CServerDlg::OnRecordingDirClicked()
+{
+    // get the current value from pServer
+    QString currentValue = pServer->GetRecordingDir();
+    QString newRecordingDir = QFileDialog::getExistingDirectory ( this,
+                                                                  tr ( "Select Main Recording Directory" ),
+                                                                  currentValue,
+                                                                  QFileDialog::ShowDirsOnly | QFileDialog::DontUseNativeDialog );
+    if ( newRecordingDir != currentValue )
+    {
+        pServer->SetRecordingDir ( newRecordingDir );
+        UpdateRecorderStatus ( QString::null );
+    }
+}
+
+void CServerDlg::OnClearRecordingDirClicked()
+{
+    if ( pServer->GetRecorderErrMsg() != QString::null || pServer->GetRecordingDir() != "" )
```

Comment:
Why do you mix QString::null and ""?

Reply:
The recording directory is an empty string because it should never be null.

The error message being null means "there is no error message" explicitly - not that the error message is an empty string.